src/Mvc/Mvc.RazorPages/src/Builder/PageActionEndpointConventionBuilder.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Builds conventions that will be used for customization of <see cref="EndpointBuilder"/> instances.
+    /// </summary>
+    /// <remarks>
+    /// This interface is used at application startup to customize endpoints for the application.
+    /// </remarks>
+    public sealed class PageActionEndpointConventionBuilder : IEndpointConventionBuilder
+    {
+        private readonly object _lock;
+        private readonly List<Action<EndpointBuilder>> _conventions;
+
+        internal PageActionEndpointConventionBuilder(object @lock, List<Action<EndpointBuilder>> conventions)
+        {
+            _lock = @lock;
```

Comment:
This is a little unusual. I think its worth a brief comment explaining where the lock is coming from